python/pyspark/sql/dataframe.py

```diff
@@ -470,7 +470,7 @@ def sampleBy(self, col, fractions, seed=None):
         +---+-----+
 
         """
-        if not isinstance(col, str):
+        if not isinstance(col, basestring):
```

Comment:
I agree with you. There is no problem caused by allowing unicode here.
As you mentioned, it's better to handle `'a'` and `u'a'` because there are few cases that unicode is passed. (e.g. when `__future__.unicode_literals` is imported in Python 2.)